fix(api): remove partiallyCompletedChallenges and savedChallenges from get-public-profile endpoint

ZunedKhan07 opened the pull request:
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes locally on my machine.

Closes #57378

This PR adds the missing test instruction to step 27 of the JavaScript Algorithms and Data Structures certification project.

It also ensures that the `partiallyCompletedChallenges` and `savedChallenges` fields are no longer included in the public profile response from the API.

ilenia-magoni commented:
the issue you are linking is alredy closed, and doesn't seem to match the PR you have opened, please clarify what these changes are for

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/workshop-fruit-search-app/step-5.md:
why is this change included in this PR?

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lecture-understanding-html-attributes/66f6db08d55022680a3cfbc9.md:
why is this change included in this PR?

ilenia-magoni commented on client/src/components/Progress/progress-bar.tsx:
why is this change included in this PR?

ZunedKhan07 commented:
I initially started working based on issue #57378, but it seems the PR got linked to #61649 automatically. The changes I made are based on the original instructions in #57378 and its discussion thread.

ZunedKhan07 commented:
Done with the cleanup!
Removed partiallyCompletedChallenges and savedChallenges from the get-public-profile endpoint as per #61649.
Also restored unrelated changes and removed extra test files.
Ready for another review — please let me know if anything else is required. 🙏

ilenia-magoni commented:
now there are more files that need to be removed

majestic-owl448 reviewed (commented):
please try again, the files that are not related to the PR are still in the PR

you should reach out on discord (https://chat.freecodecamp.org) or the forum (https://forum.freecodecamp.org) for help

ilenia-magoni commented on curriculum/challenges/english/25-front-end-development/lecture-understanding-html-attributes/66f6db08d55022680a3cfbc9.md:
now this file is being deleted, please instead remove any changes to it

ZunedKhan07 commented:
Thanks for the feedback! I’ve removed the unintended `i18n-curriculum` folder from the PR. Now the only change is the test file: `project-player-removal.test.js`.

Please let me know if anything else needs to be adjusted. 😊

ZunedKhan07 commented:
I've now restored the `.gitmodules` file as requested. Let me know if any further changes are needed.

ilenia-magoni commented on xs:
this new file appeared, it should not be here

ilenia-magoni commented on curriculum/i18n-curriculum:
this is still in the PR

ojeytonwilliams commented:
Hi @ZunedKhan07 there's one commit that contains the relevant changes: https://github.com/freeCodeCamp/freeCodeCamp/pull/61649/commits/1cc3c9b4dac36cfd334754c9d1dcde5bcce482bf

The simplest thing to do would be to drop all other commits and force push just that one.

ZunedKhan07 commented:
Thanks @ojeytonwilliams — I have reset the branch and force pushed only the relevant commit (`1cc3c9b`) as suggested.

ilenia-magoni commented:
make sure your main branch is on par with the main branch of this fork, and that there are not extra commits, so you can create clean branches from the main branch and you will not have the extra files anymore

look at the second point here: https://contribute.freecodecamp.org/how-to-contribute-to-the-codebase/#contributing-to-the-codebase
